File: JavaClient/src/ai/AI.java
package ai;

import team.koala.chillin.client.RealtimeAI;
import ks.KSObject;
import ks.models.*;
import ks.commands.*;


public class AI extends RealtimeAI<World, KSObject> {

	public AI(World world) {
		super(world);
	}

	@Override
	public void initialize() {
		System.out.println("initialize");
	}

	@Override
	public void decide() {
		System.out.println("decide");

		if (this.mySide.equals("Police")) {
	        for (Police police : this.world.getPolices())
	        {
	            if (police.getStatus() == EAgentStatus.Dead)
	                continue;

	            boolean doingBombOperation = police.getDefusionRemainingTime() != -1;
	            if (doingBombOperation)
	            {
	                System.out.println("Agent[" + police.getId() + "]: " + "Continue Bomb Operation");
	                continue;
	            }

	            ECommandDirection bombsiteDirection = findBombsiteDirection(police.getPosition());
	            if (bombsiteDirection == null)
	            {
	            	System.out.println("Agent[" + police.getId() + "]: " + "Random Move");
	            	int randIndex = (int) (Math.random() * ECommandDirection.values().length);
	            	ECommandDirection randDir = ECommandDirection.values()[randIndex];
	                move(police.getId(), randDir);
	            }
	            else
	            {
	            	System.out.println("Agent[" + police.getId() + "]: " + "Start Bomb Operation");
	                defuse(police.getId(), bombsiteDirection);
	            }
	        }
	    }
	    else
	    {
	    	for (Terrorist terrorist : this.world.getTerrorists())
	        {
	            if (terrorist.getStatus() == EAgentStatus.Dead)
	                continue;

	            boolean doingBombOperation = terrorist.getPlantingRemainingTime() != -1;
	            if (doingBombOperation)
	            {
	                System.out.println("Agent[" + terrorist.getId() + "]: " + "Continue Bomb Operation");
	                continue;
	            }

	            ECommandDirection bombsiteDirection = findBombsiteDirection(terrorist.getPosition());
	            if (bombsiteDirection == null)
	            {
	            	System.out.println("Agent[" + terrorist.getId() + "]: " + "Random Move");
	            	int randIndex = (int) (Math.random() * ECommandDirection.values().length);
	            	ECommandDirection randDir = ECommandDirection.values()[randIndex];
	                move(terrorist.getId(), randDir);
	            }
	            else
	            {
	            	System.out.println("Agent[" + terrorist.getId() + "]: " + "Start Bomb Operation");
	                plant(terrorist.getId(), bombsiteDirection);
	            }
	        }
	    }
	}


	public void move(int agentId, ECommandDirection moveDirection)
	{
		this.sendCommand(new Move() {{ id = agentId; direction = moveDirection; }});
	}

	public void plant(int agentId, ECommandDirection bombsiteDirection)
	{
		this.sendCommand(new PlantBomb() {{ id = agentId; direction = bombsiteDirection; }});
	}

	public void defuse(int agentId, ECommandDirection bombsiteDirection)
	{
		this.sendCommand(new DefuseBomb() {{ id = agentId; direction = bombsiteDirection; }});
	}


	private ECommandDirection findBombsiteDirection(Position position)
	{
	    if ((this.world.getBoard().get(position.getY() - 1).get(position.getX()).getValue() >= ECell.SmallBombSite.getValue()) &&
	        (this.world.getBoard().get(position.getY() - 1).get(position.getX()).getValue() <= ECell.VastBombSite.getValue()))
	        return ECommandDirection.Up;

	    if ((this.world.getBoard().get(position.getY()).get(position.getX() + 1).getValue() >= ECell.SmallBombSite.getValue()) &&
	        (this.world.getBoard().get(position.getY()).get(position.getX() + 1).getValue() <= ECell.VastBombSite.getValue()))
	        return ECommandDirection.Right;
	
	    if ((this.world.getBoard().get(position.getY() + 1).get(position.getX()).getValue() >= ECell.SmallBombSite.getValue()) &&
	        (this.world.getBoard().get(position.getY() + 1).get(position.getX()).getValue() <= ECell.VastBombSite.getValue()))
	        return ECommandDirection.Down;
	
	    if ((this.world.getBoard().get(position.getY()).get(position.getX() - 1).getValue() >= ECell.SmallBombSite.getValue()) &&
	        (this.world.getBoard().get(position.getY()).get(position.getX() - 1).getValue() <= ECell.VastBombSite.getValue()))
	        return ECommandDirection.Left;
	
	    return null;
	}
}

File: CSharpClient/Game/AI.cs
using System;

using KoalaTeam.Chillin.Client;
using KS;
using KS.Commands;
using KS.Models;

using KSObject = KS.KSObject;

namespace Game
{
	public class AI : RealtimeAI<World, KSObject>
	{
		private readonly Random random = new Random();

		public AI(World world) : base(world)
		{
		}

		public override void Initialize()
		{
			Console.WriteLine("initialize");
		}

		public override void Decide()
		{
			Console.WriteLine("decide");

			if (this.MySide == "Police")
			{
				foreach (var police in this.World.Polices)
				{
					if (police.Status == EAgentStatus.Dead)
						continue;

					bool doingBombOperation = police.DefusionRemainingTime != -1;
					if (doingBombOperation)
					{
						Console.WriteLine("Agent[" + police.Id + "]: " + "Continue Bomb Operation");
						continue;
					}

					ECommandDirection? bombsiteDirection = FindBombsiteDirection(police.Position);
					if (bombsiteDirection == null)
					{
						Console.WriteLine("Agent[" + police.Id + "]: " + "Random Move");
						var randDir = (ECommandDirection)random.Next(Enum.GetNames(typeof(ECommandDirection)).Length);
						Move(police.Id, randDir);
					}
					else
					{
						Console.WriteLine("Agent[" + police.Id + "]: " + "Start Bomb Operation");
						Defuse(police.Id, bombsiteDirection.Value);
					}
				}
			}
			else
			{
				foreach (var terrorist in this.World.Terrorists)
				{
					if (terrorist.Status == EAgentStatus.Dead)
						continue;

					bool doingBombOperation = terrorist.PlantingRemainingTime != -1;
					if (doingBombOperation)
					{
						Console.WriteLine("Agent[" + terrorist.Id + "]: " + "Continue Bomb Operation");
						continue;
					}

					ECommandDirection? bombsiteDirection = FindBombsiteDirection(terrorist.Position);
					if (bombsiteDirection == null)
					{
						Console.WriteLine("Agent[" + terrorist.Id + "]: " + "Random Move");
						var randDir = (ECommandDirection)random.Next(Enum.GetNames(typeof(ECommandDirection)).Length);
						Move(terrorist.Id, randDir);
					}
					else
					{
						Console.WriteLine("Agent[" + terrorist.Id + "]: " + "Start Bomb Operation");
						Plant(terrorist.Id, bombsiteDirection.Value);
					}
				}
			}
		}


		public void Move(int? agentId, ECommandDirection moveDirection)
		{
			this.SendCommand(new Move() { Id = agentId, Direction = moveDirection });
		}

		public void Plant(int? agentId, ECommandDirection bombsiteDirection)
		{
			this.SendCommand(new PlantBomb() { Id = agentId, Direction = bombsiteDirection });
		}

		public void Defuse(int? agentId, ECommandDirection bombsiteDirection)
		{
			this.SendCommand(new DefuseBomb() { Id = agentId, Direction = bombsiteDirection });
		}


		private ECommandDirection? FindBombsiteDirection(Position position)
		{
			if ((World.Board[(int)position.Y - 1][(int)position.X].Value >= ECell.SmallBombSite) &&
				(World.Board[(int)position.Y - 1][(int)position.X].Value <= ECell.VastBombSite))
				return ECommandDirection.Up;

			if ((World.Board[(int)position.Y][(int)position.X + 1].Value >= ECell.SmallBombSite) &&
				(World.Board[(int)position.Y][(int)position.X + 1].Value <= ECell.VastBombSite))
				return ECommandDirection.Right;

			if ((World.Board[(int)position.Y + 1][(int)position.X].Value >= ECell.SmallBombSite) &&
				(World.Board[(int)position.Y + 1][(int)position.X].Value <= ECell.VastBombSite))
				return ECommandDirection.Down;

			if ((World.Board[(int)position.Y][(int)position.X - 1].Value >= ECell.SmallBombSite) &&
				(World.Board[(int)position.Y][(int)position.X - 1].Value <= ECell.VastBombSite))
				return ECommandDirection.Left;

			return null;
		}
	}
}
